src/main.rs

```diff
@@ -1,6 +1,30 @@
-// Copyright 2021 IOTA Stiftung
+// Copyright 2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
+use actix::{
+    Actor,
+    System,
+};
+use chronicle::{
+    api::{
+        ChronicleAPI,
+        ShutdownAPI,
+    },
+    config::mongo::{
+        Credential,
+        MongoConfig,
+    },
+};
+
 fn main() {
-    println!("Hello world from Chronicle!");
+    dotenv::dotenv().unwrap();
+    env_logger::init();
+    let system = System::new();
+    let mongo_config =
+        MongoConfig::default().with_credential(Credential::default().with_username("root").with_password("pass"));
```

Comment:
I think we should go with the `MongoConfig::new().with_xxx(...).finish()` paradigm here, because it gives us the flexibility to add additional logic in the `.finish()` method if needed (for example for validation).

I think the way we handle the configuration of the node in Bee is quite sensible: https://github.com/iotaledger/bee/blob/shimmer-develop/bee-node/bee-node/src/config.rs

Reply:
Yes, but the bee Node config is actually a builder pattern in the sense that all fields are required in the final struct. Our config struct only exists to mirror the `ClientOptions` struct from mongodb, because it cannot be Serialized (for some reason...), and it contains entirely optional fields. It makes more sense to use the `MongoConfig` struct itself as the builder and simply return the `ClientOptions` if requested in the `build/finish` step.